feed-rs/src/parser/tests.rs

```diff
@@ -0,0 +1,55 @@
+use crate::model::Link;
+use crate::parser;
+use crate::parser::{generate_id_from_link_and_title, util};
+use crate::util::test;
+
+// Regression test for the default ID generator
+#[test]
+fn id_generator_default() {
+    let test_data = test::fixture_as_raw("rss2/rss_2.0_kdist.xml");
+    let feed = parser::parse(test_data.as_slice()).unwrap();
+    assert_eq!("354331764be7571efc15c7a1bad13d54", feed.id);
+}
+
+// Custom implementation providing backward compatibility with v0.2
+#[test]
+fn id_generator_v0_2() {
+    let test_data = test::fixture_as_raw("rss2/rss_2.0_kdist.xml");
+
+    // Custom ID that trims URLs etc
+    let feed = parser::Builder::new()
+        .id_generator(|links, title, _uri| {
+            // If we have a link without relative components, use that
+            if let Some(link) = links.iter().find(|l| l.rel.is_none()) {
+                // Trim the trailing slash if it exists
+                let mut link = Link::new(link.href.clone(), None);
```

Comment:
this is great, but this *only works* feed-rs/src, since Link::new() is private, so this means copying out the whole new() implementation *and* parse_uri() which is also private :v

*and also* importing the uuid crate. this should just be a self-contained `pub fn feed_rs_0_2_id_generator()` and you can keep everything else private.

Reply:
Ah, of course ... I didn't think of this. I moved the the test to integration, and refactor it so the v0.2 generator is a method on the builder in #223